### PR TITLE
Checking Storage and StorageTest

### PR DESCRIPTION
**Pull request by Sergei-Nerobeev**
1.  Add class Storage
2.  Class Storage contains methods adding and taking 4 type of cars.
3.  Add class StorageTest for testing methods of class Storage.

### CODE REVIEW

File: src/hu/nero/toyota/storage/Storage.java [hunk 1]
```diff
@@ -1,42 +1,155 @@
...
-    totalCars -= 1;
-    maxCars += 1;
-    System.out.println(
-        "Taken from Storage: " + takenCamry + ". Total cars: " + totalCars + "." + " Free place: " + maxCars);
-    return takenCamry;
-
-  }*/
-
-  }
+    private int maxNumberOfCars = 1000; // макс количество машин на складе
```
Comment (danikirillov) [inline src/hu/nero/toyota/storage/Storage.java:9]: make it final and static

File: src/hu/nero/toyota/storage/Storage.java [hunk 2]
```diff
@@ -1,42 +1,155 @@
...
+    private int totalCarsInMainStorage; // общее количество машин на складе
+    private int totalCamryStorage;
+    private int totalSolaraStorage;
+    private int totalDynaStorage;
+    private int totalHianceStorage;
+    private final Object[][] mainStorage;
```
Comment (danikirillov) [inline src/hu/nero/toyota/storage/Storage.java:15]: instead of 2 dims array, create 4 arrays

File: src/hu/nero/toyota/storage/Storage.java [hunk 3]
```diff
@@ -1,42 +1,155 @@
...
+    public Storage() {
+        this.totalCarsInMainStorage = 0;
+        this.mainStorage = new Object[1][maxNumberOfCars];
+        Camry[] camry = new Camry[totalCamryStorage];
+        Solara[] solara = new Solara[totalSolaraStorage];
+        Dyna[] dyna = new Dyna[totalDynaStorage];
+        Hiance[] hiance = new Hiance[totalHianceStorage];
+    }
+
+    public void counterOfCamry() {
```
Comment (danikirillov) [inline src/hu/nero/toyota/storage/Storage.java:26]: change the name

File: src/hu/nero/toyota/storage/Storage.java [hunk 4]
```diff
@@ -1,42 +1,155 @@
...
+        System.out.println("Count of Dyna in Dyna Storage now are: " + result);
+    }
+
+    public void counterOfHiance() {
+        int result = getTotalHianceStorage();
+        System.out.println("Count of Hiance in Hiance Storage now are: " + result);
+    }
+
+    public Camry addCamry(Camry camry) {
+        this.mainStorage[0][0] = camry;
```
Comment (danikirillov) [inline src/hu/nero/toyota/storage/Storage.java:46]: save not in 0, but in totalCamryStorage

File: src/hu/nero/toyota/storage/Storage.java [hunk 5]
```diff
@@ -1,42 +1,155 @@
...
+        totalCarsInMainStorage++;
+        totalCamryStorage++;
+        maxNumberOfCars--;
+        System.out.println("Added to to Camry Storage: " + camry + ". Total cars on the Storage: " + totalCarsInMainStorage
+                + ". Free place for: " + maxNumberOfCars);
+
+        return camry;
+    }
+
+    public Camry takenCamry() {
```
Comment (danikirillov) [inline src/hu/nero/toyota/storage/Storage.java:56]: take

File: src/hu/nero/toyota/storage/Storage.java [hunk 6]
```diff
@@ -1,42 +1,155 @@
...
+        Camry takenCamry = (Camry) this.mainStorage[0][0];
+        totalCarsInMainStorage--;
+        totalCamryStorage--;
+        maxNumberOfCars++;
```
Comment (danikirillov) [inline src/hu/nero/toyota/storage/Storage.java:60]: what will happen if the storage is empty?

File: src/hu/nero/toyota/test/StorageTest.java [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+import hu.nero.toyota.storage.Storage;
+
+public class StorageTest {
+    public static void main(String[] args) throws CountryFactoryNotEqualException {
+        testStorageCreation();
+        testAddCamry();
+        testTakenCamry();
+    }
+
+    static void testStorageCreation() throws CountryFactoryNotEqualException {
```
Comment (danikirillov) [inline src/hu/nero/toyota/test/StorageTest.java:18]: create a test for 10 cars creation and check that totalCarsInMainStorage is correct, and after that remove 5 and check totalCarsInMainStorage is correct.

Also create test and try to get a car from an empty storage

**Review by danikirillov — COMMENTED** [review]
👍

File: src/hu/nero/toyota/storage/Storage.java (hunk 2 above)
Comment (Sergei-Nerobeev) [reply src/hu/nero/toyota/storage/Storage.java:15]: Fixed!

File: src/hu/nero/toyota/storage/Storage.java (hunk 3 above)
Comment (Sergei-Nerobeev) [reply src/hu/nero/toyota/storage/Storage.java]: Fixed!

File: src/hu/nero/toyota/storage/Storage.java [hunk 8]
```diff
@@ -1,42 +1,94 @@
...
-    totalCars -= 1;
-    maxCars += 1;
-    System.out.println(
-        "Taken from Storage: " + takenCamry + ". Total cars: " + totalCars + "." + " Free place: " + maxCars);
-    return takenCamry;
-
-  }*/
-
-  }
+    private static int maxNumberOfCars = 1000; // макс количество машин на складе
```
Comment (danikirillov) [inline src/hu/nero/toyota/storage/Storage.java:11]: please stop resolving comments, which are written by someone else.
Instead of resolving - write fixed. (or make all changes and ask me to review again:) 

You forgot to make the field final.

If you did it intentionally, explain why.

File: src/hu/nero/toyota/storage/Storage.java (hunk 4 above)
Comment (Sergei-Nerobeev) [reply src/hu/nero/toyota/storage/Storage.java]: Fixed

File: src/hu/nero/toyota/storage/Storage.java (hunk 5 above)
Comment (Sergei-Nerobeev) [reply src/hu/nero/toyota/storage/Storage.java]: Fixed!

File: src/hu/nero/toyota/storage/Storage.java [hunk 9]
```diff
@@ -1,42 +1,155 @@
...
+        totalCamryStorage++;
+        maxNumberOfCars--;
+        System.out.println("Added to to Camry Storage: " + camry + ". Total cars on the Storage: " + totalCarsInMainStorage
+                + ". Free place for: " + maxNumberOfCars);
+
+        return camry;
+    }
+
+    public Camry takenCamry() {
+        Camry takenCamry = (Camry) this.mainStorage[0][0];
```
[the remark this answers is not in the document]
Comment (Sergei-Nerobeev) [reply src/hu/nero/toyota/storage/Storage.java]: Fixed

File: src/hu/nero/toyota/storage/Storage.java (hunk 6 above)
Comment (Sergei-Nerobeev) [reply src/hu/nero/toyota/storage/Storage.java]: Fixed!

File: src/hu/nero/toyota/storage/Storage.java (hunk 8 above)
Comment (Sergei-Nerobeev) [reply src/hu/nero/toyota/storage/Storage.java]: Fixed!

File: src/hu/nero/toyota/test/StorageTest.java (hunk 7 above)
Comment (Sergei-Nerobeev) [reply src/hu/nero/toyota/test/StorageTest.java]: Fixed!
